src/deadline/unreal_perforce_utils/secret_manager.py

```
@@ -0,0 +1,91 @@
+
+from deadline.unreal_logger import get_logger
+
+
+logger = get_logger()
+
+
+def get_secret_manager_client():
+    """
+    Creates a boto3 client for the AWS SecretsManager service.
+
+    This function figures out the region to use. If the environment variable
+    `AWS_REGION_NAME` is set, it will be used. Otherwise, the region will be
+    determined by the `botocore.utils.InstanceMetadataRegionFetcher.retrieve_region()`.
+
+    :return: A SecretsManager client
+    """
+
+    client_params = {"service_name": "secretsmanager"}
+
+    if "AWS_REGION_NAME" in os.environ and os.environ["AWS_REGION_NAME"] != "":
+        client_params["region_name"] = os.environ["AWS_REGION_NAME"]
+    else:
+        fetcher = InstanceMetadataRegionFetcher()
+        client_params["region_name"] = fetcher.retrieve_region()
+    logger.info(f"SecretsManager client parameters: {client_params}")
+
+    return boto3.client(**client_params)
+
+
+def get_secret_from_env(secret_variable_name: str) -> Optional[str]:
+    """
+    Retrieves a secret from Boto3 SecretsManager using passed environment variable name to get secret name.
+
+    :param secret_variable_name: The name of the environment variable containing the secret id
+    :type secret_variable_name: str
+    :return: The secret string
+    :rtype: Optional[str]
+    """
+    logger.info(f"Getting secret from environment variable: {secret_variable_name}")
+
+    sm_client = get_secret_manager_client()
+
+    secret_id = os.getenv(secret_variable_name)
+    if secret_id in [None, ""]:
+        logger.warning(f"Cant get secret from empty environment variable {secret_variable_name}")
+        return None
+
+    try:
+        response = sm_client.get_secret_value(SecretId=secret_id)
+    except ClientError as e:
+        logger.warning(
+            f"Failed to get secret from Boto3 SecretsManager by id {secret_id}."
+            f"{e} -- {traceback.format_exc()}"
+        )
+        return None
```

Same comment from above, should we raise an exception here?

---

hmm. Maybe yes. But for that we need to check if AWS_SECRET_P4INFO in os environ on the top. And if it is in, we expect that we should retrieve something from SM. Good point, will update